### ISSUE TITLE
Automation "Touch" mode not working in Reaper

### ISSUE THREAD

**Issue by alfonso73** [pierreguillot/Camomile#219]
Hi!
My enviroment : Windows 10 (64bit), Reaper 6.18dev1212 (64bit), Camomile 1.07 VST3, Sample Rate 44.1 kHz, ASIO buffer 64 samples

Steps to reproduce : 
Load a Camomile VST3 plugin and try to automate any of its parameters using Reaper automation "Touch" mode.

Observed Results : 
Except (sometimes) the very first automated parameter the parameters doesn't respond to automation and stay at their level.

Expected Result : 
The parameters should respond to their movement/automation when automated in "Touch" mode.
BTW in all the other automation modes they respond as expected.

I've found similar problems with other Juce based VST3 plugins as for example here...
https://github.com/asb2m10/dexed/issues/244

Here's something that might give some clue
https://forum.juce.com/t/touch-automation-toggles-beginparameterchangegesture/21528/23

Here's a video showing the issue https://youtu.be/JYpSg-vJfLc
(the constant rhythmic glitch you here is related to buffering issue in OBS recording so nothing related to Reaper or Camomile)
ohh...and i realized that Camomile VST3 isn't visible beacause i'm video recording the reaper's windows only. But you can see the parameters being automated while i'm moving Camomile's sliders i put i my patch.

**Comment by alfonso73** [comment]
hi @pierreguillot! Do you have any idea on what's goin' on with Camomile and Reaper's parameter automation in "Touch" mode?
thanks
ps...i'm trying to build some VST using else libraries and as soon as i resolve some polyphony issues i'll share it.

**Comment by pierreguillot** [comment]
No, I didn't have the time to test yet... sorry. I'll try to do it soon.

**Comment by alfonso73** [comment]
Thanks for your reply @pierreguillot. Let me know when and if you need some testing when you'll have the time to look into it. best!

**Comment by pierreguillot** [comment]
I tested on macOS with the plugin AlmondOrgan and it seems to be okay. I'll test on Windows later today. 

Can you ensure that the plugin is not muted and so on? Because if the DAW doesn't call the audio callback of the plugin, the messages might not be sent to the plugin and retrieved. 

Is there any error message in the Camomile console? And can you share a basic patch that reproduces the problem?

**Comment by pierreguillot** [comment]
I tested on Windows and there is no problem either.

**Comment by alfonso73** [comment]
Hi @pierreguillot. Yes you're right. I just realized that i was messing with the whole VST3 parameter management in my pd patch. I've adopted your [gui.link] abstraction and your smart naming convention ($0-parametername, $0-parametername-sld-s, $0-parametername-sld-r, $0-parametername-nbx-s, $0-parametername-nbx-r). Everything works fine. 
thanks!
PS i wonder if this repo is the correct place for make some request about your Camomile examples. I don't want to clutter your developing space.
